core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -167,7 +167,12 @@ private static String unencodeResourceURIToFilePath(@NotNull final String resour
     private String resolvePath() {
         String result = getResourcePath();
 
-        if (SystemUtils.IS_OS_WINDOWS && result.startsWith("/")) {
+        // Special cases for Windows
+        if (SystemUtils.IS_OS_WINDOWS && result.matches("^/[A-Z]:/dev/.*")) {
```

Comment:
I don't particularly like having to add this special case behaviour - may move elsewhere.

This prevents `/dev/shm` eventually becoming `/host_mnt/c/dev/shm` on Docker for Windows. If this broken path is used for the SHM mount then SHM is unusable inside the container; the path needs to not be remapped onto `/host_mnt/c`, and should remain `/dev/shm`.

Reply:
Actually.... what if we avoid using `addFileSystemBind` (that [creates a mountable file](https://github.com/testcontainers/testcontainers-java/blob/c9b66eb3e610bfb76024000ae062bec889cdab34/core/src/main/java/org/testcontainers/containers/GenericContainer.java#L734)) but use the bind API directly for the SHM?

This ~~hack~~ adjustment does solve the problem too, but I think we're fixing it in the wrong place. WDYT?

Reply:
That's a good point - I did think of something similar but didn't want to extend our API just for this. What I didn't think of was just doing `this.getBinds().add( ... )`, which I suppose would work.

Will give it a try.